tests/test_metrics.py

```diff
@@ -97,6 +97,67 @@ def test(self):
         self.assertTrue(ret["baz"]["metric_hcsv"] == ["baz"])
 
 
+class TestMetricsRecursive(TestDvc):
+    def setUp(self):
+        super(TestMetricsRecursive, self).setUp()
+        self.dvc.scm.commit("init")
+
+        self.dvc.scm.checkout("nested", create_new=True)
+
+        os.mkdir("nested")
+        os.mkdir(os.path.join("nested", "subnested"))
+
+        ret = main(
+            [
+                "run",
+                "-M",
+                os.path.join("nested", "metric_nested"),
+                "echo",
```

Comment:
Looks like `echo` on windows doesn't know about `-n` option :) Could we try without it? You might want to make `_read_metric` strip what it reads before showing https://github.com/iterative/dvc/blob/master/dvc/project.py#L1023 .